database-plugins/src/main/java/co/cask/hydrator/plugin/DBRecord.java

```diff
@@ -150,6 +150,13 @@ public void write(PreparedStatement stmt) throws SQLException {
   private Object transformValue(int sqlColumnType, Object original) throws SQLException {
     if (original != null) {
       switch (sqlColumnType) {
+        case Types.SMALLINT:
+        case Types.TINYINT:
+          if (original.getClass() == Short.class) {
```

Comment:
you should be able to do ((Number) original).intValue() instead of checking the class